Source/JavaScriptCore/runtime/JSGlobalObject.cpp

```diff
@@ -2633,6 +2634,9 @@ void JSGlobalObject::visitChildrenImpl(JSCell* cell, Visitor& visitor)
     thisObject->m_typedArrayProto.visit(visitor);
     thisObject->m_typedArraySuperConstructor.visit(visitor);
     thisObject->m_regExpGlobalData.visitAggregate(visitor);
+
+    for (auto& object : thisObject->m_objectsForDeferredTaskTimer)
+        visitor.append(object);
```

Comment:
Let's take a cellLock in this scope. GC marker can run concurrently.

Reply:
How do these references ever get cleared?

Reply:
Once the JSGlobalObject is no longer needed by others, then only this remains. After that, the GC will consider they are no longer referenced and then collected.

```
JSGlobalObject -> JSObjectsForDeferredTaskTimer
        ^                               |
        |                                 |
        -------------------
```